aws/resource_aws_codebuild_report_group.go

```diff
@@ -0,0 +1,283 @@
+									},
+									"packaging": {
+										Type:         schema.TypeString,
+										Optional:     true,
+										Default:      codebuild.ReportPackagingTypeNone,
+										ValidateFunc: validation.StringInSlice(codebuild.ReportPackagingType_Values(), false),
+									},
+									"path": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsCodeBuildReportGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+	createOpts := &codebuild.CreateReportGroupInput{
+		Name:         aws.String(d.Get("name").(string)),
+		Type:         aws.String(d.Get("type").(string)),
+		ExportConfig: expandAwsCodeBuildReportGroupExportConfig(d.Get("export_config").([]interface{})),
+		Tags:         keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().CodebuildTags(),
+	}
+
+	resp, err := conn.CreateReportGroup(createOpts)
+	if err != nil {
+		return fmt.Errorf("error creating CodeBuild Report Groups: %w", err)
+	}
+
+	d.SetId(aws.StringValue(resp.ReportGroup.Arn))
+
+	return resourceAwsCodeBuildReportGroupRead(d, meta)
+}
+
+func resourceAwsCodeBuildReportGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	resp, err := conn.BatchGetReportGroups(&codebuild.BatchGetReportGroupsInput{
+		ReportGroupArns: aws.StringSlice([]string{d.Id()}),
+	})
+	if err != nil {
+		return fmt.Errorf("error Listing CodeBuild Report Groups: %w", err)
+	}
+
+	if len(resp.ReportGroups) == 0 {
+		log.Printf("[WARN] CodeBuild Report Groups (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	reportGroup := resp.ReportGroups[0]
+
+	if reportGroup == nil {
+		log.Printf("[WARN] CodeBuild Report Groups (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("arn", reportGroup.Arn)
+	d.Set("type", reportGroup.Type)
+	d.Set("name", reportGroup.Name)
+
+	if err := d.Set("created", reportGroup.Created.Format(time.RFC3339)); err != nil {
+		return fmt.Errorf("error setting created: %w", err)
+	}
+
+	if err := d.Set("export_config", flattenAwsCodeBuildReportGroupExportConfig(reportGroup.ExportConfig)); err != nil {
+		return fmt.Errorf("error setting export config: %w", err)
+	}
+
+	if err := d.Set("tags", keyvaluetags.CodebuildKeyValueTags(reportGroup.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	return nil
+}
+
+func resourceAwsCodeBuildReportGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	input := &codebuild.UpdateReportGroupInput{
+		Arn: aws.String(d.Id()),
+	}
+
+	updateNeeded := false
```

Comment:
Is this necessary? `resourceAwsCodeBuildReportGroupUpdate()` won't be called unless at least one of `export_config` or `tags` has changed.

Reply:
your right, removing